pkg/karmadactl/cmdinit/kubernetes/deployments.go

```diff
@@ -268,7 +269,7 @@ func (i *CommandInitOption) makeKarmadaKubeControllerManagerDeployment() *appsv1
 					"--controllers=namespace,garbagecollector,serviceaccount-token",
 					"--kubeconfig=/etc/kubeconfig",
 					"--leader-elect=true",
-					fmt.Sprintf("--leader-elect-resource-namespace=%s", i.Namespace),
+					fmt.Sprintf("--leader-elect-resource-namespace=%s", util.NamespaceKarmadaSystem),
```

Comment:
Should this value also be configurable?

Reply:
Maybe it should be. I don't know the exact scope of influence.

As far as I know, The related code is as follows:

https://github.com/karmada-io/karmada/blob/0b73c6b4e9c84a4ef702f9f7af87d6bfb038fb56/pkg/karmadactl/cmdinit/karmada/deploy.go#L42-L49

https://github.com/karmada-io/karmada/blob/0b73c6b4e9c84a4ef702f9f7af87d6bfb038fb56/pkg/karmadactl/cmdinit/karmada/deploy.go#L167-L183

https://github.com/karmada-io/karmada/blob/0b73c6b4e9c84a4ef702f9f7af87d6bfb038fb56/pkg/karmadactl/cmdinit/karmada/deploy.go#L191-L211

The above code is about hard coded namespace `karmada-system` in karmada which is created during karmada init.

https://github.com/karmada-io/karmada/blob/0b73c6b4e9c84a4ef702f9f7af87d6bfb038fb56/pkg/karmadactl/cmdinit/kubernetes/deployments.go#L271

https://github.com/karmada-io/karmada/blob/0b73c6b4e9c84a4ef702f9f7af87d6bfb038fb56/pkg/karmadactl/cmdinit/kubernetes/deployments.go#L392

The above code need the namespace which should be created in karmada during init, but now the namespace isn't created in karmada.

If the value should be configurable, maybe we can replace the hard coded namespace `karmada-system`？e.g.

https://github.com/karmada-io/karmada/blob/0b73c6b4e9c84a4ef702f9f7af87d6bfb038fb56/pkg/karmadactl/cmdinit/karmada/deploy.go#L28

If you think it's OK, I can try to make the change.

Reply:
I think what you're saying is quite reasonable. 

These are actually two different changes, can you take them apart?

Reply:
OK, then if this PR can be merged, I'll make some change base on it to replace the hard coded namespace `karmada-system` in karmada.

Reply:
I can't get why we need to hard code the namespace here.
It's beyond the scope of this PR, right? (If yes, I recommend filing another PR for it.)

Reply:
ping @sayaoailun

Reply:
> I can't get why we need to hard code the namespace here. It's beyond the scope of this PR, right? (If yes, I recommend filing another PR for it.)

OK, new PR after this PR.

This PR is about custom namespace in host cluster.

Next PR is about the hard coded namespace in Karmada.

Reply:
What's the benefit to hard-coding the `--leader-elect-resource-namespace`? Or, what's the problem if we put the leader election resource under the namespace `i.Namespace`?

I think this PR is focusing on solving `-n` doesn't work issue, is it the mandatory part of the solution?

Reply:
> Or, what's the problem if we put the leader election resource under the namespace `i.Namespace`?

For now, the namespace `i.Namespace` was not created in Karmada but the hard coded namespace `karamda-system`. When  `i.Namespace` is not set as `karamda-system`, `KarmadaKubeControllerManager` and `KarmadaScheduler` couldn't create the lease object.